### DISCUSSION TITLE
Connecting DAQ channels to devices

### DISCUSSION THREAD

**Discussion by adamkglaser** [AllenNeuralDynamics/aind-data-schema#478] [General]
I feel like this was discussed many months ago, and I cannot remember the conclusion we came to. This is coming back up, because I am updating the ExA-SPIM instrument JSON generation, and realized there is only a soft connection between the ```device_name``` input for ```DAQChannel``` and the actual devices.

What would be the best way to reference ```device_name```? Should there be some sort of ```name``` attribute for all devices?

@saskiad @dyf

**Comment by saskiad** [comment]
yes, this is the purpose of the name attribute in `Device`. We made it optional at someone's request, but it's there for exactly this purpose. I'm happy to make it required if that would help

**Comment by adamkglaser** [comment]
Got it ok there is a name attribute. That is exactly what I was interested in.
Adam

From: Saskia de Vries ***@***.***>
Date: Tuesday, September 12, 2023 at 10:30 AM
To: AllenNeuralDynamics/aind-data-schema ***@***.***>
Cc: adamkglaser ***@***.***>, Author ***@***.***>
Subject: Re: [AllenNeuralDynamics/aind-data-schema] Connecting DAQ channels to devices (Discussion #478)

yes, this is the purpose of the name attribute in Device. We made it optional at someone's request, but it's there for exactly this purpose. I'm happy to make it required if that would help

**Comment by dyf** [comment]
Cleaning up the board - looks like this is closed.
